packages/core/contracts/zodiac/OptimisticOracleModule.sol

```diff
@@ -0,0 +1,197 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.6;
+
+import "@gnosis.pm/zodiac/contracts/core/Module.sol";
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import "../oracle/implementation/Constants.sol";
+import "../oracle/interfaces/FinderInterface.sol";
+import "../oracle/interfaces/SkinnyOptimisticOracleInterface.sol";
+import "../oracle/interfaces/OracleAncillaryInterface.sol";
+
+contract OptimisticOracleModule is Module {
+    event OptimisticOracleModuleDeployed(address indexed owner, address indexed avatar, address target);
+
+    event TransactionsProposed(uint256 indexed proposalId, address indexed proposer, uint256 indexed proposalTime);
+
+    event TransactionExecuted(uint256 indexed proposalId, uint256 indexed transactionIndex);
+
+    event ProposalDeleted(uint256 indexed proposalId);
+
+    FinderInterface public finder;
+    SkinnyOptimisticOracleInterface optimisticOracle;
+
+    IERC20 public collateral;
+    uint256 public bond;
+    string public rules;
```

Comment:
What does rules do? Is there some interface that requires it or something?

Reply:
This is a reference to the off-chain published rules that the OO should consider when determining if a proposal is valid or not. The UMIP draft has more details. 

https://github.com/UMAprotocol/UMIPs/pull/492